### PR TITLE
Added forge worlds compatibility

### PR DESCRIPTION
**Pull request by Angel-0**
This pull request adds a compatibility for forge worlds. That means you can change your skin in every world (even forge ones).

### CODE REVIEW

**Comment by Angel-0** [comment]
I apologize for those many commits, I kinda messed up with my tests

File: src/main/java/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java [hunk 1]
```diff
@@ -116,11 +117,8 @@ public void updateSkin(Player player) {
             Object difficulty = ReflectionUtil.invokeMethod(world, "getDifficulty");
             Object worlddata = ReflectionUtil.getObject(world, "worldData");
             Object worldtype = ReflectionUtil.invokeMethod(worlddata, "getType");
+            World.Environment p_e = player.getWorld().getEnvironment();
```
Comment (McLive) [inline src/main/java/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java]:
```suggestion
            World.Environment environment = player.getWorld().getEnvironment();
```

File: src/main/java/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java [hunk 2]
```diff
@@ -129,12 +127,17 @@ public void updateSkin(Player player) {
 
             Object respawn = null;
             try {
+                dimension = p_e.getId();
```
Comment (McLive) [inline src/main/java/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java]:
```suggestion
                dimension = environment.getId();
```

File: src/main/java/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java [hunk 3]
```diff
@@ -129,12 +127,17 @@ public void updateSkin(Player player) {
...
                 respawn = ReflectionUtil.invokeConstructor(PlayOutRespawn,
                         new Class<?>[]{
                                 int.class, PEACEFUL.getClass(), worldtype.getClass(), enumGamemode.getClass()
                         },
                         dimension, difficulty, worldtype, ReflectionUtil.invokeMethod(enumGamemode.getClass(), null, "getById", new Class<?>[]{int.class}, gmid));
             } catch (Exception ignored) {
+                if (p_e.equals(World.Environment.NETHER))
+                    dimension = -1;
+                else if (p_e.equals(World.Environment.THE_END))
```
Comment (McLive) [inline src/main/java/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java]:
```suggestion
                if (environment.equals(World.Environment.NETHER))
                    dimension = -1;
                else if (environment.equals(World.Environment.THE_END))
```

File: src/main/java/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java [hunk 4]
```diff
@@ -276,7 +279,12 @@ public void updateSkin(Player player) {
                     sendPacket(playerCon, addInfo);
                     sendPacket(playerCon, respawn);
 
-                    ReflectionUtil.invokeMethod(craftHandle, "updateAbilities");
+                    try {
+                        ReflectionUtil.invokeMethod(craftHandle, "updateAbilities");
+
+                    } catch (InvocationTargetException e) {
+                        e.getCause().printStackTrace();
+                    }
```
Comment (McLive) [inline src/main/java/skinsrestorer/bukkit/skinfactory/UniversalSkinFactory.java]:
```suggestion
                    ReflectionUtil.invokeMethod(craftHandle, "updateAbilities");
```
This should only be used for debugging purpose

**Pull request merged**